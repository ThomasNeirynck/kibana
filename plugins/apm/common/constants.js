export const APP_NAME = 'context.app.name';
export const APP_AGENT_NAME = 'context.app.agent.name';
export const EVENT_PROCESSOR_NAME = 'processor.name';

export const TRANSACTION_DURATION = 'transaction.duration.us';
export const TRANSACTION_TYPE = 'transaction.type';
export const TRANSACTION_RESULT = 'transaction.result';
export const TRANSACTION_NAME = 'transaction.name';
export const TRANSACTION_ID = 'transaction.id';

export const TRACE_START = 'trace.start.us';
export const TRACE_DURATION = 'trace.duration.us';
export const TRACE_TRANSACTION_ID = 'trace.transaction_id';
export const TRACE_TYPE = 'trace.type';
export const TRACE_NAME = 'trace.name';
export const TRACE_ID = 'trace.id';
export const TRACE_SQL = 'context.db.statement';

export const ERROR_GROUP_ID = 'error.grouping_key';
export const ERROR_CULPRIT = 'error.culprit';
export const ERROR_MESSAGE = 'error.exception.message';
