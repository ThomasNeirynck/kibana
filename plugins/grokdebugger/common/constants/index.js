export { ROUTES } from './routes';
export { PLUGIN } from './plugin';
