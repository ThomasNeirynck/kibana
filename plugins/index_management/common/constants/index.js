export { PLUGIN } from './plugin';
export { BASE_PATH } from './base_path';
export * from './index_statuses';
