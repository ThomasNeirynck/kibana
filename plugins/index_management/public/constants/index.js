export { REFRESH_RATE_INDEX_LIST } from './refresh_intervals';
