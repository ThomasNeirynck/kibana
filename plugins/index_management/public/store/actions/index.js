export * from './clear_cache_indices';
export * from './close_indices';
export * from './delete_indices';
export * from './flush_indices';
export * from './forcemerge_indices';
export * from './load_indices';
export * from './load_index_data';
export * from './open_indices';
export * from './refresh_indices';
export * from './reload_indices';
export * from './table_state';
export * from './edit_index_settings';
export * from './update_index_settings';
export * from './detail_panel';

