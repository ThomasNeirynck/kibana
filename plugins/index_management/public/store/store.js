import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';

import { indexManagement } from './reducers/';

export const indexManagementStore = (initialState = {}) => {
  const enhancers = [ applyMiddleware(thunk) ];

  window.__REDUX_DEVTOOLS_EXTENSION__ && enhancers.push(window.__REDUX_DEVTOOLS_EXTENSION__());
  return createStore(
    indexManagement,
    initialState,
    compose(...enhancers)
  );
};
