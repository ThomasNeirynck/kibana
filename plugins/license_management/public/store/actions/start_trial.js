import { createAction } from 'redux-actions';
import { canStartTrial, startTrial } from '../../lib/es';
import { toastNotifications } from 'ui/notify';

export const trialStatusLoaded = createAction(
  'LICENSE_MANAGEMENT_TRIAL_STATUS_LOADED'
);

export const loadTrialStatus = () => async dispatch => {
  const trialOK = await canStartTrial();
  dispatch(trialStatusLoaded(trialOK));
};

export const startLicenseTrial = () => async (
  dispatch,
  getState,
  { xPackInfo }
) => {
  /*eslint camelcase: 0*/
  const { trial_was_started, error_message } = await startTrial();
  if (trial_was_started) {
    await xPackInfo.refresh();
    // reload necessary to get left nav to refresh with proper links
    window.location.reload();
  } else {
    return toastNotifications.addDanger(error_message);
  }
};
