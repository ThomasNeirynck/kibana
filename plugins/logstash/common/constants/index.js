export { INDEX_NAMES } from './index_names';
export { TYPE_NAMES } from './type_names';
export { ROUTES } from './routes';
export { PAGINATION } from './pagination';
export { PLUGIN } from './plugin';
