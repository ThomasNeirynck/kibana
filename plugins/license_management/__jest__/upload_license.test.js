import TestRenderer from 'react-test-renderer';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import Wrapper from '../public/app';
import { Provider } from 'react-redux';
import { uploadLicense } from "../public/store/actions/upload_license";
import { licenseManagementStore } from "../public/store/store";
import { BASE_PATH } from '../common/constants';

import sinon from 'sinon';

let server = null;
let store = null;
let component = null;
const services = {
  kbnUrl: {
    change: jest.fn()
  },
  autoLogout: () => {},
  xPackInfo: {
    refresh: jest.fn(),
    get: () => { return { license: { type: 'basic' } }; }
  }
};

describe('UploadLicense', () => {
  beforeEach(() => {
    server = sinon.fakeServer.create();
    server.respondImmediately = true;
    store = licenseManagementStore();
    component = (
      <Provider store={store}>
        <MemoryRouter initialEntries={[ `${BASE_PATH}upload_license`]}>
          <Wrapper />
        </MemoryRouter>
      </Provider>
    );
  });
  afterEach(() => {
    server.restore();
    services.xPackInfo.refresh.mockReset();
    services.kbnUrl.change.mockReset();
  });
  it('should display an error when submitting invalid JSON', async () => {
    store.dispatch(uploadLicense('INVALID', 'trial'));
    const rendered = TestRenderer.create(component);
    expect(rendered).toMatchSnapshot();
  });
  it('should display an error when ES says license is invalid', async () => {
    const invalidLicense = JSON.stringify({ license: { type: 'basic' } });
    server.respond([200, { "Content-Type": "application/json" },
      '{"acknowledged": "true", "license_status": "invalid"}']);
    await uploadLicense(invalidLicense)(store.dispatch, null, services);
    const rendered = TestRenderer.create(component);
    expect(rendered).toMatchSnapshot();
  });
  it('should display an error when ES says license is expired', async () => {
    const invalidLicense = JSON.stringify({ license: { type: 'basic' } });
    server.respond([200, { "Content-Type": "application/json" },
      '{"acknowledged": "true", "license_status": "expired"}']);
    await uploadLicense(invalidLicense)(store.dispatch, null, services);
    const rendered = TestRenderer.create(component);
    expect(rendered).toMatchSnapshot();
  });
  it('should display a modal when license requires acknowledgement', async () => {
    const unacknowledgedLicense = JSON.stringify({ license: { type: 'basic' } });
    /* eslint-disable max-len */
    server.respond([200, { "Content-Type": "application/json" },
      `{
          "acknowledged":false,
          "license_status":"valid",
          "acknowledge":
            {
              "message": "This license update requires acknowledgement. To acknowledge the license, please read the following messages and update the license again, this time with the \\"acknowledge=true\\" parameter:",
              "watcher":["Watcher will be disabled"]
            }
          }`]);
    /* eslint-enable max-len */
    await uploadLicense(unacknowledgedLicense,  'trial')(store.dispatch, null, services);
    const rendered = TestRenderer.create(component);
    expect(rendered).toMatchSnapshot();
  });
  it('should refresh xpack info and navigate to BASE_PATH when ES accepts new license', async () => {
    const validLicense = JSON.stringify({ license: { type: 'basic' } });
    server.respond([200, { "Content-Type": "application/json" },
      '{"acknowledged": "true", "license_status": "valid"}']);
    await uploadLicense(validLicense)(store.dispatch, null, services);
    expect(services.xPackInfo.refresh).toHaveBeenCalled();
    expect(services.kbnUrl.change).toHaveBeenCalledWith(BASE_PATH);
  });
  it('should display error when ES returns error', async () => {
    const license = JSON.stringify({ license: { type: 'basic' } });
    /* eslint-disable max-len */
    server.respond([200, { "Content-Type": "application/json" },
      `{
        "error":
        {
          "root_cause":
            [{
              "type":"illegal_state_exception",
              "reason":"Can not upgrade to a production license unless TLS is configured or security is disabled"
            }],"type":"illegal_state_exception",
            "reason":"Can not upgrade to a production license unless TLS is configured or security is disabled"},
            "status":500}
      `]);
  /* eslint-enable max-len */
    await uploadLicense(license)(store.dispatch, null, services);
    const rendered = TestRenderer.create(component);
    expect(rendered).toMatchSnapshot();
  });
});

