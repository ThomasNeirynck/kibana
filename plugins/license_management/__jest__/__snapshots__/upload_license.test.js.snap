// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadLicense should display a modal when license requires acknowledgement 1`] = `
<div
  className="kuiNotice"
>
  <div
    className="kuiPanel"
  >
    <div
      className="kuiModalOverlay"
    >
      <div>
        <div
          className="kuiModal kuiModal--confirmation"
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <div
            className="kuiModalHeader"
          >
            <div
              className="kuiModalHeader__title"
              data-test-subj="confirmModalTitleText"
            >
              Confirm License Upload
            </div>
          </div>
          <div
            className="kuiModalBody"
          >
            <div
              data-test-subj="confirmModalBodyText"
            >
              <div>
                <p>
                  Some functionality will be lost if you replace your 
        TRIAL license with a
        BASIC license.  Review the list of features below.
                </p>
                <ul>
                  <li>
                    Watcher will be disabled
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            className="kuiModalFooter"
          >
            <button
              autoFocus={false}
              className="kuiButton kuiButton--hollow"
              data-test-subj="confirmModalCancelButton"
              onClick={[Function]}
            >
              <span
                className="kuiButton__inner"
              >
                <span>
                  Cancel
                </span>
              </span>
            </button>
            <button
              autoFocus={false}
              className="kuiButton kuiButton--primary"
              data-test-subj="confirmModalConfirmButton"
              onClick={[Function]}
            >
              <span
                className="kuiButton__inner"
              >
                <span>
                  Confirm
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="kuiPanelHeader"
    >
      <div
        className="kuiPanelHeader__title"
      >
        Upload a license
      </div>
    </div>
    <div
      className="kuiPanelBody"
    >
      <p>
        Your license key is a JSON file with a signature attached.
      </p>
      <p>
        Uploading a license will replace your current 
        <b>
          
        </b>
         license.
      </p>
      <form>
        <div
          className="kuiVerticalRhythm"
        >
          <p>
            <input
              name="licenseFile"
              onChange={[Function]}
              type="file"
            />
          </p>
        </div>
        <button
          className="kuiButton kuiButton--primary"
          disabled={false}
          onClick={[Function]}
        >
          <span
            className="kuiButton__inner"
          >
            <span>
              Upload
            </span>
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`UploadLicense should display an error when ES says license is expired 1`] = `
<div
  className="kuiNotice"
>
  <div
    className="kuiPanel"
  >
    <div
      className="kuiPanelHeader"
    >
      <div
        className="kuiPanelHeader__title"
      >
        Upload a license
      </div>
    </div>
    <div
      className="kuiPanelBody"
    >
      <p>
        Your license key is a JSON file with a signature attached.
      </p>
      <p>
        Uploading a license will replace your current 
        <b>
          
        </b>
         license.
      </p>
      <form>
        <div
          className="kuiInfoPanel kuiInfoPanel--error kuiVerticalRhythm"
        >
          <div
            className="kuiInfoPanelHeader"
          >
            <span
              aria-label="Warning"
              className="kuiInfoPanelHeader__icon kuiIcon kuiIcon--error fa-warning"
              role="img"
            />
            <span
              className="kuiInfoPanelHeader__title"
            >
              The supplied license has expired.
            </span>
          </div>
        </div>
        <div
          className="kuiVerticalRhythm"
        >
          <p>
            <input
              name="licenseFile"
              onChange={[Function]}
              type="file"
            />
          </p>
        </div>
        <button
          className="kuiButton kuiButton--primary"
          disabled={false}
          onClick={[Function]}
        >
          <span
            className="kuiButton__inner"
          >
            <span>
              Upload
            </span>
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`UploadLicense should display an error when ES says license is invalid 1`] = `
<div
  className="kuiNotice"
>
  <div
    className="kuiPanel"
  >
    <div
      className="kuiPanelHeader"
    >
      <div
        className="kuiPanelHeader__title"
      >
        Upload a license
      </div>
    </div>
    <div
      className="kuiPanelBody"
    >
      <p>
        Your license key is a JSON file with a signature attached.
      </p>
      <p>
        Uploading a license will replace your current 
        <b>
          
        </b>
         license.
      </p>
      <form>
        <div
          className="kuiInfoPanel kuiInfoPanel--error kuiVerticalRhythm"
        >
          <div
            className="kuiInfoPanelHeader"
          >
            <span
              aria-label="Warning"
              className="kuiInfoPanelHeader__icon kuiIcon kuiIcon--error fa-warning"
              role="img"
            />
            <span
              className="kuiInfoPanelHeader__title"
            >
              The supplied license is not valid for this product.
            </span>
          </div>
        </div>
        <div
          className="kuiVerticalRhythm"
        >
          <p>
            <input
              name="licenseFile"
              onChange={[Function]}
              type="file"
            />
          </p>
        </div>
        <button
          className="kuiButton kuiButton--primary"
          disabled={false}
          onClick={[Function]}
        >
          <span
            className="kuiButton__inner"
          >
            <span>
              Upload
            </span>
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`UploadLicense should display an error when submitting invalid JSON 1`] = `
<div
  className="kuiNotice"
>
  <div
    className="kuiPanel"
  >
    <div
      className="kuiPanelHeader"
    >
      <div
        className="kuiPanelHeader__title"
      >
        Upload a license
      </div>
    </div>
    <div
      className="kuiPanelBody"
    >
      <p>
        Your license key is a JSON file with a signature attached.
      </p>
      <p>
        Uploading a license will replace your current 
        <b>
          
        </b>
         license.
      </p>
      <form>
        <div
          className="kuiInfoPanel kuiInfoPanel--error kuiVerticalRhythm"
        >
          <div
            className="kuiInfoPanelHeader"
          >
            <span
              aria-label="Warning"
              className="kuiInfoPanelHeader__icon kuiIcon kuiIcon--error fa-warning"
              role="img"
            />
            <span
              className="kuiInfoPanelHeader__title"
            >
              Error encountered uploading license: Check your license file.
            </span>
          </div>
        </div>
        <div
          className="kuiVerticalRhythm"
        >
          <p>
            <input
              name="licenseFile"
              onChange={[Function]}
              type="file"
            />
          </p>
        </div>
        <button
          className="kuiButton kuiButton--primary"
          disabled={false}
          onClick={[Function]}
        >
          <span
            className="kuiButton__inner"
          >
            <span>
              Upload
            </span>
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`UploadLicense should display error when ES returns error 1`] = `
<div
  className="kuiNotice"
>
  <div
    className="kuiPanel"
  >
    <div
      className="kuiPanelHeader"
    >
      <div
        className="kuiPanelHeader__title"
      >
        Upload a license
      </div>
    </div>
    <div
      className="kuiPanelBody"
    >
      <p>
        Your license key is a JSON file with a signature attached.
      </p>
      <p>
        Uploading a license will replace your current 
        <b>
          
        </b>
         license.
      </p>
      <form>
        <div
          className="kuiInfoPanel kuiInfoPanel--error kuiVerticalRhythm"
        >
          <div
            className="kuiInfoPanelHeader"
          >
            <span
              aria-label="Warning"
              className="kuiInfoPanelHeader__icon kuiIcon kuiIcon--error fa-warning"
              role="img"
            />
            <span
              className="kuiInfoPanelHeader__title"
            >
              Error encountered uploading license: Can not upgrade to a production license unless TLS is configured or security is disabled
            </span>
          </div>
        </div>
        <div
          className="kuiVerticalRhythm"
        >
          <p>
            <input
              name="licenseFile"
              onChange={[Function]}
              type="file"
            />
          </p>
        </div>
        <button
          className="kuiButton kuiButton--primary"
          disabled={false}
          onClick={[Function]}
        >
          <span
            className="kuiButton__inner"
          >
            <span>
              Upload
            </span>
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;
